### PR TITLE
Workflows for automated releases

### PR DESCRIPTION
**Pull request by stephanielee9**
Updated / added three files:

1. `main.yaml` - added a job to release after a successful deploy on main
2. `ci.yml` - added workflow_call (so it can be used by other workflows) and ignored updates to `_version.py` (so the action isn't run twice - it'll already be called by the new workflow below)
3. `automated_publish_release.yml` - once `_version.py` is pushed to main, will run checks, if they pass will then run the `main.yaml` script. If the checks or deploy fails, someone can manually run `main.yaml` to try to deploy and publish release again.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -6,8 +6,11 @@ name: cirq-superstaq
 on:
   push:
     branches: [main]
+    paths-ignore:
```
Comment (dowusu-antwi) [inline .github/workflows/ci.yml:9]: This is just to avoid running the checks twice right?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -6,8 +6,11 @@ name: cirq-superstaq
...
+      - 'cirq_superstaq/_version.py'
   pull_request:
     branches: [main]
+  workflow_call: null
```
Comment (dowusu-antwi) [inline .github/workflows/ci.yml:13]: What is this line for? (Same question for line 5 in `main.yaml`)

Comment (stephanielee9) [reply .github/workflows/ci.yml:13]: `workflow_call` allows the workflow to be called by other workflows. In this case, the automated release workflow (`automated_publish_release.yml`) is just a workflow triggered when updates to `_version.py` are made to run the checks workflow and then the deploy and release workflow, if the checks pass.

Comment (dowusu-antwi) [reply .github/workflows/ci.yml:13]: Makes sense, thanks!

File: .github/workflows/automated_publish_release.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+      - 'cirq_superstaq/_version.py'
+
+jobs:
+  run_checks:
+    uses: ./.github/workflows/ci.yml
+    secrets: inherit
+  publish_and_release:
+    needs: run_checks
+    uses: ./.github/workflows/main.yaml
+    secrets: inherit
```
Comment (dowusu-antwi) [inline .github/workflows/automated_publish_release.yml:22]: @richrines1 @singular-value We've seen this before but I forgot how it was resolved -- adding `\n`? (The other `.yaml`s don't have it though...)

**Review by dowusu-antwi — DISMISSED** [review]
lgtm, pending nit

File: .github/workflows/automated_publish_release.yml (hunk 3 above)
Comment (stephanielee9) [reply .github/workflows/automated_publish_release.yml]: There is a more sophisticated way to do this that I think @vtomole knows about, but I usually just go in GitHub and manually enter a new line (hit Enter key, delete any auto-indents GitHub adds).

Comment (dowusu-antwi) [reply .github/workflows/automated_publish_release.yml]: Fixed now -- thanks @stephanielee9!

**Review by dowusu-antwi — APPROVED** [review]

**Pull request merged**